From 7d481671e6752fcde310bff8e0c1c6e77b0630ad Mon Sep 17 00:00:00 2001
From: Martin Sperl <kernel@martin.sperl.org>
Date: Tue, 17 May 2016 08:39:12 +0000
Subject: [PATCH 42/65] ASoC: hw-params-rules: iterate only over child-nodes

Used the wrong method to iterate over child nodes
(included all nodes in the dt) as there is no
for_each_child_of_node_by_name...

Signed-off-by: Martin Sperl <kernel@martin.sperl.org>
---
 sound/soc/generic/hw-params-rules.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/sound/soc/generic/hw-params-rules.c b/sound/soc/generic/hw-params-rules.c
index 2115c90..31a0d95 100644
--- a/sound/soc/generic/hw-params-rules.c
+++ b/sound/soc/generic/hw-params-rules.c
@@ -295,12 +295,14 @@ static int asoc_generic_hw_params_actionmatches_parse_of(
 	/* init matchers */
 	INIT_LIST_HEAD(list_head);
 
-	for (np = of_find_node_by_name(node, nodename); np;
-	     np = of_find_node_by_name(np, nodename)) {
-		ret = asoc_generic_hw_params_actionmatch_parse_of(
-			dev, np, list_head, nodename);
-		if (ret)
-			return ret;
+	/* iterate over all child nodes */
+	for_each_child_of_node(node, np) {
+		if (np->name && (of_node_cmp(np->name, nodename) == 0)) {
+			ret = asoc_generic_hw_params_actionmatch_parse_of(
+				dev, np, list_head, nodename);
+			if (ret)
+				return ret;
+		}
 	}
 
 	return 0;
-- 
2.7.4

